File: Arrays/Kth smallest element.py
# dumb solution by me , time complexity : O(nlogn) , expected : O(n)
class Solution:
    def kthSmallest(self,arr, l, r, k):
        arr.sort()
        return arr[k-1]
      
 # noice solution by others, time complexity: no idea
import heapq
class Solution:
    def kthSmallest(self,arr, l, r, k):
        heapq.heapify(arr)
        ans = 0
        for i in range(k):
            ans = heapq.heappop(arr)
        return ans

File: C++/Arrays/Kth smallest element.cpp
// dumb solution by me , time complexity: O(n)

class Solution{
    public:
    int kthSmallest(int arr[], int l, int r, int k) {
        
        sort(arr,arr+r+1);
        return arr[k-1];
    }
};

// noice solution by others , time complexity: O(nlogk)

// max heap is used here

int kthSmallest(int arr[], int l, int r, int k) {
//code here
priority_queue<int> mxhp;
for(int i=l;i<=r;i++){
mxhp.push(arr[i]);
if(mxhp.size()>k){
mxhp.pop();
}
}
return mxhp.top();
}
